docs/sql-ref-syntax-qry-select-cte.md

```diff
@@ -19,4 +19,120 @@ license: |
   limitations under the License.
 ---
 
-**This page is under construction**
+### Description
+
+A common table expression (CTE) defines a temporary result set that a user can reference within the duration of a SQL statement. A CTE can be used wherever a SELECT statement is used; for example, it can be used in a SELECT, INSERT, UPDATE, DELETE, MERGE or CREATE VIEW statement. A CTE has a name with an optional column names list, followed by a query expression. When a CTE references itself, it is called a recursive CTE.
```

Comment:
nit: How about this?
`A common table expression (CTE) defines a temporary result set that a user can reference within the duration of a SQL statement. `
=>
`A common table expression (CTE) defines a temporary result set that a user can reference possibly multiple times within the scope of a SQL statement. `

Reply:
Is this statement necessary? `A CTE can be used wherever a SELECT statement is used; for example, `? How about just saying `within the duration of a SQL statement. A CTE can be used in SELECT, INSERT, UPDATE, DELETE, MERGE, or CREATE VIEW statements`?

Reply:
UPDATE and MERGE are supported? https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala#L765-L768

Reply:
I guess maybe only mention the SELECT?
```A common table expression (CTE) defines a temporary result set that a user can reference possibly multiple times within the scope of a SQL statement. A CTE is used mainly in a SELECT statement. It is defined with a name including an optional column names list, followed by a query expression. When a CTE references itself, it is called a recursive CTE.```
Does this look OK to you?

Reply:
We need to describe `It is defined with a name including an optional column names list, followed by a query expression.` in the head section? [The syntax section](https://github.com/apache/spark/pull/28196/files#diff-771ee2387e08ce855bea13e43117c735R26-R35) looks enough to me.

Reply:
will remove.